### PR TITLE
GRW-1695 / Feature / GTM metadata fixes

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

- Fix inital virtual_page_view tracking
- Change the way we provide metadata so Google Tag Manager can process it correctly
- Add siteVersion context for future A/B testing data

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

File: apps/store/src/services/gtm.tsx [hunk 1]
```diff
@@ -120,3 +121,14 @@ export const useGTMEvents = () => {
     }
   }, [router.events])
 }
+
+const trackPageView = (url: string) => {
+  console.debug('pageview', url)
```
Comment (robinandeer) [inline apps/store/src/services/gtm.tsx:126]: Do we want to use console here or always datadog logs?

File: apps/store/src/services/gtm.tsx [hunk 2]
```diff
@@ -95,22 +96,22 @@ export const useGTMEvents = () => {
...
+        siteVersion: 'racoon',
         country: countryCode,
+        environment: getGtmEnvironment(),
       },
     })
   }, [countryCode])
 
+  useEffect(() => {
+    trackPageView(window.location.pathname)
+  }, [])
```
Comment (robinandeer) [inline apps/store/src/services/gtm.tsx:110]: Shouldn't this be run outside React?

```
if (typeof window === 'undefined') { ... }
```

File: apps/store/src/services/gtm.tsx (hunk 1 above)
Comment (alebedev) [reply apps/store/src/services/gtm.tsx:126]: Intentionally not sending to Datadog here, since we'll be sending that data to Analytics

Comment (alebedev) [reply apps/store/src/services/gtm.tsx:124]: I think it's benign and can stay without harming us.  For example, JIRA outputs lots of debug logs in production and that doesn't seem to be a problem

But let's figure this out closer to launch

File: apps/store/src/services/gtm.tsx (hunk 2 above)
Comment (alebedev) [reply apps/store/src/services/gtm.tsx]: Just enabling it didn't work, created a ticket to investigate and fix in the future

File: apps/store/src/pages/_app.tsx [hunk 3]
```diff
@@ -17,6 +17,8 @@ if (process.env.NEXT_PUBLIC_API_MOCKING === 'enabled') {
 
 if (typeof window === 'undefined') {
   Datadog.initDatadog()
```
[the remark this answers is not in the document]
Comment (alebedev) [reply apps/store/src/pages/_app.tsx:24]: Won't work, created backlog task to fix this

**Pull request merged**